src/middlewared/middlewared/plugins/audit/backend.py

```diff
@@ -163,28 +175,26 @@ def __fetchall(self, conn, qs):
 
         return data
 
-    @private
-    @accepts(
-        Str('db_name', enum=[svc[0] for svc in AUDITED_SERVICES], required=True),
-        Ref('query-filters'),
-        Ref('query-options')
-    )
+    @api_method(AuditBackendQueryArgs, AuditBackendQueryResult, private=True)
     def query(self, db_name, filters, options):
         """
         Query the specied auditable service's database based on the specified
         `query-filters` and `query-options`. This is the private endpoint for the
         audit backend and so it should generally not be used by websocket API
         consumers except in special circumstances.
         """
-        conn = self.connections[db_name]
+        try:
+            conn = self.connections[db_name]
+            if conn.connection is None:
+                raise CallError(
+                    f'{db_name}: connection to audit database is not initialized.'
+                )
+        except KeyError:
```

Comment:
`KeyError` must be handled only for the first line, not for the rest of the code